app/src/main/java/org/oppia/app/player/state/InteractionAdapter.kt

```diff
@@ -0,0 +1,142 @@
+class InteractionAdapter(
+  private val htmlParserFactory: HtmlParser.Factory,
+  private val entityType: String,
+  private val explorationId: String,
+  val itemList: Array<String>?,
+  private val interactionId: String
+  ) : RecyclerView.Adapter<RecyclerView.ViewHolder>(), InteractionAnswerRetriever {
+
+  private var mSelectedItem = -1
+
+  private var selectedAnswerIndex = -1
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+    return when (viewType) {
+      VIEW_TYPE_MULTIPLE_CHOICE -> {
+        val inflater = LayoutInflater.from(parent.getContext())
+        val binding =
+          DataBindingUtil.inflate<MultipleChoiceInteractionItemsBinding>(
+            inflater,
+            R.layout.multiple_choice_interaction_items,
+            parent,
+            false
+          )
+        MultipleChoiceViewHolder(binding)
+      }
+      VIEW_TYPE_ITEM_SELECTION -> {
+        val inflater = LayoutInflater.from(parent.getContext())
+        val binding =
+          DataBindingUtil.inflate<ItemSelectionInteractionItemsBinding>(
+            inflater,
+            R.layout.item_selection_interaction_items,
+            parent,
+            false
+          )
+        ItemSelectionViewHolder(binding)
+      }
+      else -> throw IllegalArgumentException("Invalid view type")
+    }
+  }
+
+  override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+    when (holder.itemViewType) {
+      VIEW_TYPE_MULTIPLE_CHOICE -> (holder as MultipleChoiceViewHolder).bind(
+        itemList!!.get(position),
+        position,
+        mSelectedItem
+      )
+      VIEW_TYPE_ITEM_SELECTION -> (holder as ItemSelectionViewHolder).bind(
+        itemList!!.get(position),
+        position,
+        mSelectedItem
+      )
+    }
+  }
+
+  // Determines the appropriate ViewType according to the interaction type.
+  override fun getItemViewType(position: Int): Int {
+    return if (interactionId.equals("ItemSelectionInput")) {
+      VIEW_TYPE_ITEM_SELECTION
+    } else {
+      VIEW_TYPE_MULTIPLE_CHOICE
+    }
+  }
+
+  override fun getItemCount(): Int {
+    return itemList!!.size
+  }
+
+  private inner class ItemSelectionViewHolder(val binding: ViewDataBinding) : RecyclerView.ViewHolder(binding.root) {
+    internal fun bind(rawString: String, position: Int, selectedPosition: Int) {
+      binding.setVariable(BR.htmlContent, rawString)
+      binding.executePendingBindings();
+      val htmlResult: Spannable = htmlParserFactory.create(entityType, explorationId).parseOppiaHtml(
+        rawString,
+        binding.root.tv_item_selection_contents
+      )
+      binding.root.tv_item_selection_contents.text = htmlResult
+
+      binding.root.checkbox_container.setOnClickListener {
+        if (binding.root.item_selection_checkbox.isChecked)
+          binding.root.item_selection_checkbox.setChecked(false)
+        else
+          binding.root.item_selection_checkbox.setChecked(true)
```

Comment:
Here and elsewhere

`binding.root.item_selection_checkbox.isChecked = true`

Make sure you solve all warnings before PR.

Reply:
Done.